gpflow/optimizers/mcmc.py

```diff
@@ -47,25 +44,24 @@ def run_chain_fn():
         parameter_samples = hmc_helper.convert_samples_to_parameter_values(hmc_samples)
 
     Args:
-        target_log_prob_fn: Python callable which represents log-density under the target distribution.
-        model_parameters: List of `tensorflow.Variable`s or `gpflow.Parameter`s used as a state of the Markov chain.
+        target_log_prob_fn: a callable which returns the log-density of the
+            model under the target distribution.
+        model_parameters: List of `gpflow.Parameter`s used as a state of the
+            Markov chain.
     """
 
     def __init__(
-        self, target_log_prob_fn: LogProbabilityFunction, model_parameters: ModelParameters
+        self, target_log_prob_fn: Callable[[], tf.Tensor], model_parameters: Sequence[Parameter]
     ):
-        assert all([isinstance(p, (Parameter, tf.Variable)) for p in model_parameters])
+        if not all(isinstance(p, Parameter) for p in model_parameters):
+            raise TypeError(
+                "SamplingHelper only accepts `gpflow.Parameter`s (wrap tf.Variable if required)"
+            )
 
```

Comment:
```suggestion
        if not all(p.prior is not None for p in model_parameters):
            logger.warning('Priors not set on all parameters')
```

Reply:
Now that we know they are all parameters, it might also be worth checking that priors have been assigned to each of them.

Reply:
would you prefer a warning for parameter but no prior set? or are you fine with just raising a ValueError as soon as anything isn't a parameter-with-prior?